File: algorithms/2_medium/S/03618_split_array_by_prime_indices/solution.py
class Solution:
    def is_prime(self, n):
        if n < 2:
            return False

        if n == 2 or n == 3 or n == 5:
            return True

        if n % 2 == 0 or n % 3 == 0 or n % 5 == 0:
            return False

        for i in range(7, int(sqrt(n)) + 1, 2):
            if n % i == 0:
                return False

        return True

    def splitArray(self, nums: List[int]) -> int:
        sum = 0
        for i in range(len(nums)):
            if self.is_prime(i):
                sum += nums[i]
            else:
                sum -= nums[i]

        return abs(sum)

File: algorithms/2_medium/S/03618_split_array_by_prime_indices/solution.rs
impl Solution {
    #[inline]
    fn is_prime(n: usize) -> bool {
        if n == 2 || n == 3 || n == 5 {
            return true;
        }

        if n < 2 || n % 2 == 0 || n % 3 == 0 || n % 5 == 0 {
            return false;
        }

        let sqrt_n = (n as f64).sqrt() as usize;

        for i in (7..=sqrt_n).step_by(2) {
            if n % i == 0 {
                return false;
            }
        }

        true
    }

    pub fn split_array(nums: Vec<i32>) -> i64 {

        let mut sum = 0_i64;

        for i in 0..nums.len() {
            if Self::is_prime(i) {
                sum += nums[i] as i64;
            } else {
                sum -= nums[i] as i64;
            }
        }

        sum.abs()
    }
}
